files/run_worker.sh

```diff
@@ -16,6 +16,22 @@ else
   LOGLEVEL="DEBUG"
 fi
 
+# find out which queues to serve, packit-worker-0 serve both queues
+SHORT_RUNNING_DEFAULT="1"
+LONG_RUNNING_DEFAULT="0"
+SHORT_RUNNING="${SHORT_RUNNING:-SHORT_RUNNING_DEFAULT}"
+LONG_RUNNING="${LONG_RUNNING:-LONG_RUNNING_DEFAULT}"
+
+WORKER_NUMBER="$(cut -d'-' -f3 <<< ${HOSTNAME})"
+
+if [ "$WORKER_NUMBER" -ge 1 ] && [ "$WORKER_NUMBER" -lt $((1 + SHORT_RUNNING)) ]; then
+  QUEUES="short-running"
+elif [ "$WORKER_NUMBER" -ge $((1 + SHORT_RUNNING)) ] && [ "$WORKER_NUMBER" -lt $((1 + SHORT_RUNNING + LONG_RUNNING)) ]; then
+  QUEUES="long-running"
+else
+  QUEUES="short-running,long-running"
+fi
```

Comment:
Nice! I hope you've tested those conditionals..;)

Reply:
Could you add a comment translating the condition into human language? Thanks

Reply:
I see this env var as configuration so why not just define `QUEUES` in the [deployment](https://github.com/packit/deployment/blob/900fbd85a2529e3c8d569bf25f1e821d033d82c7/openshift/packit-worker.yml.j2#L75) instead of calculating it here?

You could do this by splitting the current `worker_replicas` var in the deployment playbook into `workers_long_tasks` and `workers_all_tasks` (or some better names if you have an idea :smiley:), with defaults `workers_long_tasks: 0` and `workers_all_tasks: 1`, and decide based on this whether to run [a second task to deploy](https://github.com/packit/deployment/blob/900fbd85a2529e3c8d569bf25f1e821d033d82c7/playbooks/deploy.yml#L172) a set of workers with the value of `QUEUES` set only to `long-running`.

The end result in prod would be that there would be two sets of stateful sets deployed for workers each with the specified number of replicas.

Reply:
I was thinking about this option, but if we would have one set of workers for both queues and then one set of workers for short-running tasks and another one for long-running tasks, we would have 3 sets of stateful sets and the only difference between them would be that one variable...it made better sense to me to have just one stateful set and not divide it, since it will create duplication and basically all would do the same thing. On the other hand, this would make things more straightforward, so I am really not sure about it.
What do others think?

Reply:
I'd say both solutions have pros and cons: we should discuss this tomorrow on the architecture. I'm wondering if we could have the mapping definition b/w workers and queues in a file as a configuration so we wouldn't need to rebuild the image if we wanted to change the setup and at the same time, have only a single stateful set.